tk_toolchain/cmd_line_tools/tk_build_qt_resources/__init__.py

```diff
@@ -156,15 +176,21 @@ def main():
     }
 
     print("Building user interfaces...")
-    for ui_file in args.uifiles:
-        build_params["compiler"] = args.uic
-        build_params["filename"] = ui_file
-        build_qt_params = build_ui(**build_params)
+    for index, ui_file in enumerate(args.uifiles):
+        build_params_ui_files = build_params.copy()
+        build_params_ui_files["compiler"] = args.uic
+        build_params_ui_files["filename"] = ui_file
+        if args.uifilenames:
+            build_params_ui_files["py_filename"] = args.uifilenames[index]
+        build_qt_params = build_ui(**build_params_ui_files)
```

Comment:
Perhaps something like can work?

```suggestion
        build_params_ui_files = {
            **build_params,
            "compiler": args.uic,
            "filename": ui_file,
            "py_filename": args.uifilenames[index] if args.uifilenames else None,
        }
        build_qt_params = build_ui(**build_params_ui_files)
```